### ISSUE TITLE
Precision of Time Output in NetCDF Files 

### ISSUE THREAD

**Issue by maximilian-gelbrecht** [milankl/SpeedyWeather.jl#180]
The time variable in the NetCDF files is always in `Int32` precision and in hours as a unit. However, one can also have time steps at non-integer hours. 

I'd change 

https://github.com/milankl/SpeedyWeather.jl/blob/fd963f497cb1cba0d4dad4995bb8e05f6403de64/src/output.jl#L89

to also be in `output_NF`, or is there anything speaking against that?

**Comment by milankl** [comment]
Ideally we should output in integers and not in floats. My thinking behind this was that if we ever run very long simulations (1kyr) then using Float32 for seconds since X, then we get a rounding error of half an hour or more
```julia
julia> eps(Float32(60*60*24*365*1000))/3600
0.5688889f0
```
Using Int32 but for seconds since X, we can run simulations up to
```julia
julia> typemax(Int32)/3600/24/365
68.09625973490614
```
68 years before we exhaust the dynamic range. In hours since X it's 
```julia
julia> typemax(Int32)/24/365
245146.5350456621
```
so 245kyrs. I picked the latter because I thought we wouldn't output at a higher temporal resolution than 1h anyway, but you are right, if someone wants output every 2.5 hours then this output is placed on a time axis of 0,3,5,8,10 etc...

For some reason I ran into troubles simply using Int64, which looked like something was wrong with NetCDF.jl but having investigated that in a while. If you want to change that then I'd go for Int64 for the time axis in seconds since X, but check that the netcdf output works fine before merging!

**Comment by milankl** [comment]
Alternatively, Float64 in seconds since X would work too
```julia
julia> eps(Float64(60*60*24*365*10000000))
0.0625
```
so even after 1 million years, it's more than 1/10 second accurate 🤯

**Comment by milankl** [comment]
> to also be in `output_NF`, or is there anything speaking against that?

So to come back to your question, no I wouldn't do that ☝🏼 as we probably just want to output in Float32 but it's not a good format for the time axis. At the moment we have `output_NF = NF` because outputting on a full grid equivalent cannot convert between Float32 to Float64 on the fly (reminder to self to work on this), meaning that if you run a simulation currently in Float64 then the output is in Float64 too (but truncated to `keepbits`, so size-wise it doesn't actually matter).

**Comment by maximilian-gelbrecht** [comment]
Yep, will test that. But only tomorrow morning, leaving my office in a few minutes.
